### PR TITLE
Change retrieve function in production locations API

### PR DESCRIPTION
**Pull request by protsack-stephan**
Changed the endpoint `/v1/production-locations/{os_id}` to return a single object instead of list of items with the count.

### CODE REVIEW

**Review by VadimKovalenkoSNF — APPROVED** [review]
Good work, approved.

**Comment by roman-stolar** [comment]
@protsack-stephan Good to me, but do we need to update RELEASE NOTES?

**Review by vladsha-dev — CHANGES_REQUESTED** [review]
1. I’ve left comments in the changed file.
2. Could you please update the `RELEASE-NOTES.md` file to reflect the changes you made?

File: src/django/api/views/v1/production_locations.py [hunk 1]
```diff
@@ -26,39 +26,43 @@ def __init__(self, **kwargs):
...
             OpenSearchIndexNames.PRODUCTION_LOCATIONS_INDEX,
-            query_body
+            query_body,
         )
-        return Response(response)
+        locations = response.get("data", [])
+
+        if len(locations) == 0:
+            return Response(
+                data={"detail": "Production location not found!"},
```
Comment (vladsha-dev) [inline src/django/api/views/v1/production_locations.py:64]: Could you please use the same error message as specified in [the new API spec](https://opensupplyhub.github.io/open-supply-hub-api-docs/) for GET api/v1/production-locations/{os_id} endpoint?
```
{
  "message": "The location with the given id was not found."
}
```

**Review by vladsha-dev — CHANGES_REQUESTED** [review]
I left one comment.

File: src/django/api/views/v1/production_locations.py (hunk 1 above)
Comment (protsack-stephan) [reply src/django/api/views/v1/production_locations.py]: Sure, updated the message.

**Review by vladsha-dev — APPROVED** [review]
Thanks for this fix!
LGTM

**Review by mazursasha1990 — APPROVED** [review]
LGTM

**Pull request merged**